src/hivpy/demographics.py

```diff
@@ -1,19 +1,149 @@
+import warnings
+
 import numpy as np
 from pandas.api.types import CategoricalDtype
+from scipy.interpolate import interp1d
 
 SexType = CategoricalDtype(["female", "male"])
 
 
 # Default values
 # Can wrap those in something that ensures they have a description?
 FEMALE_RATIO = 0.52
+USE_STEPWISE_AGES = False
+INC_CAT = 1
+
+
+class StepwiseAgeDistribution:
+    # stepwise distributions
+    stepwise_model1 = np.array([0.18, 0.165, 0.144, 0.114, 0.09, 0.08, 0.068,
+                                0.047, 0.036, 0.027, 0.021, 0.016, 0.012])
+    stepwise_model2 = np.array([0.15, 0.13, 0.12, 0.11, 0.1, 0.09, 0.08, 0.065,
+                                0.048, 0.04, 0.03, 0.021, 0.016])
+    stepwise_model3 = np.array([0.128, 0.119, 0.113, 0.104, 0.097, 0.09, 0.081,
+                                0.074, 0.06, 0.05, 0.038, 0.026, 0.02])
+    stepwise_boundaries = np.array([-68, -55, -45, -35, -25, -15, -5, 5, 15,
+                                    25, 35, 45, 55, 65])
+
+    def __init__(self, ages, probabilities):
+        self.probability_list = probabilities
+        self.age_boundaries = ages
+
+    @classmethod
+    def selectModel(cls, inc_cat):
+        if(inc_cat == 1):
+            return cls(cls.stepwise_boundaries, cls.stepwise_model1)
+        elif(inc_cat == 2):
+            return cls(cls.stepwise_boundaries, cls.stepwise_model1)
+        else:
+            return cls(cls.stepwise_boundaries, cls.stepwise_model1)
+
+    # Generate cumulative probabilites from stepwise distribution
+    def _genCumulativeProb(self):
+        N = self.probability_list.size
+        print("IS THIS EVEN WORKING")
```

Comment:
I forgot that was in there! Well, yes and no... The function works, but the point of that line was actually to test whether I could see printed output from running pytests, which I can't. I don't know if pytests redirects it somewhere where it can be seen, but I couldn't find it.

Reply:
Yes, pytest captures (and hides) the output by default. Running with `pytest -s`  (or `pytest --capture=no`) should show the output. You can probably configure VS Code to do that too, if you're running them from within the IDE, but not sure how.